controllers/root.js

```diff
@@ -39,16 +39,23 @@ var Root = (function() {
 
             }
 
-    		IDM.check_token(auth_token, function (user_info) {
-
+    		IDM.check_token(auth_token, function (user_info) {                
+                
                 if (config.azf.enabled) {
                     
                     AZF.check_permissions(auth_token, user_info, req, function () {
+                        if (config.work_mode && config.work_mode === 'validate'){
+                            redir_request_validate(req, res, user_info);
+                        } else if (config.work_mode &&  config.work_mode === 'proxy'){
+                            redir_request(req, res, user_info);
+                        }
+                    }, function (status, e) {
 
-                        redir_request(req, res, user_info);
+                        if (status === 401 && config.work_mode && config.work_mode === 'validate') {
```

Comment:
It's enough with checking `config.work_mode === 'validate'`. If work_mode is not defined, it should work as proxy.